src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/mega/GregTechMetaTileEntity_MegaAlloyBlastSmelter.java

```diff
@@ -0,0 +1,620 @@
+    public boolean checkMachine(IGregTechTileEntity aBaseMetaTileEntity, ItemStack aStack) {
+        if(!checkPiece("main", 5, 16, 0)) return false;
+        if (hasNormalCoils) coilLevel = HeatingCoilLevel.None;
+        if (mMaintenanceHatches.size() != 1) return false;
+        if (mMufflerHatches.size() != 45) return false;
+        if (this.glassTier < 10 && !this.mEnergyHatches.isEmpty()) {
+            for (GT_MetaTileEntity_Hatch_Energy hatchEnergy : this.mEnergyHatches) {
+                if (this.glassTier < hatchEnergy.mTier) {
+                    return false;
+                }
+            }
+        }
+        return this.glassTier >= 8 || this.getExoticEnergyHatches().size() <= 0;
+    }
+    
+    @Override
+    public int getMaxEfficiency(ItemStack aStack) {
+        return 10000;
+    }
+    
+    @Override
+    public int getDamageToComponent(ItemStack aStack) {
+        return 0;
+    }
+    
+    
+    @Override
+    public boolean explodesOnComponentBreak(ItemStack aStack) {
+        return false;
+    }
+    
+    @Override
+    public void clearHatches() {
+        super.clearHatches();
+        mExoticEnergyHatches.clear();
+    }
+    
+    public double getCoilDiscount(HeatingCoilLevel lvl) {
+        if (lvl.getTier() <= 0) return 0;
+        return Math.min(0.1, (double)Math.round(((Math.log(lvl.getTier() + 3)) / log2_40)*1000)/1000);
+    }
+    
+    @Override
+    public void explodeMultiblock() {
+        super.explodeMultiblock();
+    }
+    
+    public List<GT_MetaTileEntity_Hatch> getExoticAndNormalEnergyHatchList() {
+        List<GT_MetaTileEntity_Hatch> tHatches = new ArrayList<>();
+        tHatches.addAll(mExoticEnergyHatches);
+        tHatches.addAll(mEnergyHatches);
+        return tHatches;
+    }
+    @Override
+    public boolean drainEnergyInput(long aEU) {
+        return GT_ExoticEnergyInputHelper.drainEnergy(aEU, getExoticAndNormalEnergyHatchList());
+    }
+    
+    @Override
+    public void construct(ItemStack stackSize, boolean hintsOnly) {
+        buildPiece("main", stackSize, hintsOnly, 5, 16, 0);
+    }
+    
+    @Override
+    public IStructureDefinition<GregTechMetaTileEntity_MegaAlloyBlastSmelter> getStructureDefinition() {
+        return STRUCTURE_DEFINITION;
+    }
+    
+    @Override
+    protected GT_Multiblock_Tooltip_Builder createTooltip() {
+        GT_Multiblock_Tooltip_Builder tt = new GT_Multiblock_Tooltip_Builder();
+        tt.addMachineType("Fluid Alloy Cooker")
+        .addInfo("Controller block for the Mega Alloy Blast Smelter")
+        .addInfo("Runs the same recipes as the normal ABS, except with up to "+EnumChatFormatting.BOLD+EnumChatFormatting.UNDERLINE+MAX_PARALLELS+EnumChatFormatting.RESET+EnumChatFormatting.GRAY+" parallels.")
+                .addInfo("Every coil tier above cupronickel grants a speed bonus, based on this function:")
+                .addInfo("log(TIER - 3) / log(2^40), rounded to the nearest 10-thousandth. This caps out at 10%.")
```

Comment:
@chochem I see what you mean, it would be easier to just make it so that a linear function ends at the correct maximum value that would be the better solution so that the earlier coil tiers are not a larger upgrade than the later ones. @MadMan310  please fix this

@chochem  I see what you mean with that disagreement now, thank you for pointing out exactly where  I did not notice that the +- was wrong, however the resolution of this formula will require a rework of this anyways.

Reply:
btw, log(2^40) = 40*log(2), which is less confusing for people uncomfortable with maths.

Reply:
Please be clear on which base log as well, it appears Math.log is log base e, not base 2 or 10. In which case, use ln not log in descriptions.

Reply:
@boubou19 I don't see any reason to remove the BW dependency. You are concerned about a circular dependency, yet BW and GT++ had no dependencies on each other in the first place: https://github.com/GTNewHorizons/bartworks/blob/ff8a1e590bea7b78215a21c8f90c9955aa2ebb4c/dependencies.gradle